framework/src/executioners/Transient.C

```diff
@@ -442,8 +466,42 @@ Transient::solveStep(Real input_dt)
   // Update warehouse active objects
   _problem.updateActiveObjects();
 
+  if (_prev_time == _time && _relax_factor != 1.0) // i.e. _pic_it > 0
+  {
+    NonlinearSystem & _nl_system = _fe_problem.getNonlinearSystem();
+    System & libmesh_nl_system = _nl_system.system();
+    NumericVector<Number> & solution = *libmesh_nl_system.solution;
+    NumericVector<Number> & relax_previous = libmesh_nl_system.get_vector("relax_previous");
```

Comment:
Generally, you want to use as much MOOSE API as possible and for these 2 things we have it.

I understand you need the libMesh system for the `AllLocalDofIndicesThread`, but that class needs to be fixed to take MOOSE system rather than libMesh system. But, that should not be part of your PR.